kai/server.py

```diff
@@ -497,15 +505,38 @@ async def get_incident_solutions_for_file(request: Request):
                         "incident_uri"
                     ]
 
-        prompt = build_prompt(
-            CONFIG_IBM_GRANITE_MF,
-            {
-                "src_file_name": request_json["file_name"],
-                "src_file_language": src_file_language,
-                "src_file_contents": updated_file,
-                "incidents": incidents,
-            },
-        )
+        config_template = ""
+        base_path = os.path.dirname(__file__)
+        with open(os.path.join(base_path, "config.toml"), "rb") as f:
+            config = tomllib.load(f)
+        if "template" in config["models"]["args"]:
+            config_template = config["models"]["args"]["template"]
+
+        args = {
+            "src_file_name": request_json["file_name"],
+            "src_file_language": src_file_language,
+            "src_file_contents": updated_file,
+            "incidents": incidents,
+        }
+
+        if config["models"]["provider"].lower() == "IBMOpenSource".lower():
+            if config_template == "preamble_only":
+                prompt = build_prompt(CONFIG_IBM_LLAMA_MF_PREAMBLE_ONLY, args)
+            elif config_template == "preamble_with_analysis_only":
+                prompt = build_prompt(
+                    CONFIG_IBM_LLAMA_MF_PREAMBLE_WITH_ANALYSIS_ONLY, args
+                )
+            else:
+                prompt = build_prompt(CONFIG_IBM_LLAMA_MF, args)
+        else:
+            if config_template == "preamble_only":
+                prompt = build_prompt(CONFIG_IBM_GRANITE_MF_PREAMBLE_ONLY, args)
+            elif config_template == "preamble_with_analysis_only":
+                prompt = build_prompt(
+                    CONFIG_IBM_GRANITE_MF_PREAMBLE_WITH_ANALYSIS_ONLY, args
+                )
+            else:
+                prompt = build_prompt(CONFIG_IBM_GRANITE_MF, args)
```

Comment:
Might be better to put this in the model provider classes. Then we can just call `build_prompt(provider.get_model_config(), args)`

Reply:
Thank you for the patch @JonahSussman !